stdlib/public/Python/Python.swift

```diff
@@ -1257,6 +1257,24 @@ extension PythonObject : MutableCollection {
   }
 }
 
+extension PythonObject : Sequence {
+  public struct Iterator : IteratorProtocol {
+    fileprivate let pythonIterator: PythonObject
+
+    public func next() -> PythonObject? {
+      guard let result = PyIter_Next(pythonIterator.borrowedPyObject) else {
```

Comment:
I think this can be simplified to use ‘Python.next()’ directly. Have you tried that?

Reply:
There’s of course some trade off between efficiency and convenience. I think consistency is key here—we should either use pure C APIs for ‘makeIterator’ and ‘next’, or pure Python APIs.

Reply:
In fact, that was my first commit. I think it is better to stick with Python APIs, rather than mix and match.

Reply:
> The problem with Python API is that Python.next and iter will return PythonObject rather than PythonObject? This complicates things, because the error is thrown when the method is called dynamically.

This is okay--you can check whether the python object is equal to `Python.None`. However, I like using C APIs for both since that's strictly more efficient :)